tailscale/tailscaled-entrypoint.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+set -euxo pipefail
+
+if [[ "$(id -u)" -eq 0 ]]; then
+  mkdir -p /workspaces/.tailscale || true
+  /usr/local/sbin/tailscaled \
+    --state=/workspaces/.tailscale/tailscaled.state \
```

Comment:
I'd recommend setting --statedir=/workspaces/.tailscale/ instead of --state=..., for the benefit of features like tailscale cert and future features which use the directory.

The tailscaled.service systemd file you might have referenced pre-dates addition of the `--state=` argument for tailscaled. As `/var/lib/tailscale/` is the default this hasn't mattered, but changing it to a different path would matter.